integration/integration.go

```diff
@@ -3,28 +3,83 @@ package integration
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io"
+	"os"
+	"reflect"
 	"sync"
 
+	"github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/newrelic/infra-integrations-sdk/log"
 	"github.com/newrelic/infra-integrations-sdk/persist"
 	"github.com/pkg/errors"
 )
 
+const protocolVersion = "2"
+
 // Integration defines the format of the output JSON that integrations will return for protocol 2.
 type Integration struct {
-	locker             sync.Locker
-	storer             persist.Storer
 	Name               string    `json:"name"`
 	ProtocolVersion    string    `json:"protocol_version"`
 	IntegrationVersion string    `json:"integration_version"`
 	Entities           []*Entity `json:"data"`
+	locker             sync.Locker
+	storer             persist.Storer
 	prettyOutput       bool
 	writer             io.Writer
+	logger             log.Logger
+	args               interface{} // UGLY
```

Comment:
If we consider this isn't good code, I would add a `TODO` note about fixing it. Please, remove the `UGLY` note.